### ISSUE TITLE
Windows x32 8u202-b08 Hotspot JDK/JRE build is missing file jre/bin/client/jvm.dll

### ISSUE THREAD

**Issue by mclay** [AdoptOpenJDK/openjdk-build#871]
**Platform:**

Windows

**Architecture:**

x32

The following build obtained from the link below is missing the file jre/bin/client/jvm.dll.

https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u202-b08/OpenJDK8U-jre_x86-32_windows_hotspot_8u202b08.zip

**Comment by sxa555** [comment]
Extra info:

From  from @mclay on Slack: "the packages from oracle/azul contains both the `jre/bin/server/jvm,dll` and the `jre/bin/client/jvm.dll`"
From @sunnythepooh "There is no client hotspot for 64bit"

**Comment by karianna** [comment]
Hmm, I wonder if we need to build with a client flag somewhere.

@mclay Any reason why you're looking for a client build vs a server build?

**Comment by mclay** [comment]
@karianna It was just an observation when migrating from an Oracle x32 JRE 8 Build from 2017 to the mentioned adopt build (and a subsequent comparison with the packages from Oracle and Azul)

**Comment by Capa83** [comment]
Dear all,

when do you think we will have a fix for this?

Thanks and regards,
Tamas

**Comment by karianna** [comment]
No timeframe - pull requests are welcome on this one!

**Comment by vtaparia** [comment]
> Hmm, I wonder if we need to build with a client flag somewhere.
> 
> @mclay Any reason why you're looking for a client build vs a server build?

We are building Application to work specifically for client OS that too Windows 10 32 bit and also our application for 32 bit contains only 32 bit binaries.

**Comment by smehrbrodt** [comment]
This missing client dll does indeed cause issues.

In LibreOffice the JRE is run inside the LibreOffice process and in the 32bit process AdoptOpenJDK doesn't take long to exit with `java.lang.OutOfMemoryError: unable to create new native thread`.

The Oracle JRE could live much longer until this occured.

**Comment by smehrbrodt** [comment]
There is a configure option "--jvm-variant" which lets you choose whether you want to build the server or client variant (defaults to server).

**Comment by karianna** [comment]
@ali-ince Another on your plate.

**Comment by PaulHammes** [comment]
Dear all,

we even have the same issue. We tested the same workflow with AdoptOpenJDK and Oracle and the Oracle and get to the same result as smehrbrodt. The Orcale JRE lives much longer until getting an OutOfMemory.
We will be pleased to get a solution.

Thx and regards
Paul

**Comment by JScheiterbauer** [comment]
Hi all!

We have the same issue. In our product we also need the client JVM.
We just switched from Oracle JVM to AdoptOpenJDK because of the new licensing policy and cannot ship the changes to the customer before this is fixed.

Can you please provide the client jvm for at least the newest jdk8 builds?

Thank you very much!
Johann

**Comment by karianna** [comment]
@JScheiterbauer and @PaulHammes - Please see our support options at https://adoptopenjdk.net/support.html if you'd like to get this fix expedited.

**Comment by JScheiterbauer** [comment]
@karianna 
does that mean that if this issue is fixed right now, we would get it with 1.8.0_222 on 16th July 2019?
or do you want to say that we shall contact IBM or jClarity for commercial support and otherwise nothing will happen?

**Comment by karianna** [comment]
It will likely get fixed by the community in time for 8u222 yes (it is in the April Milestone bucket which means it's in the top 40 or so issues), if you'd like a faster fix (in a nightly build that you can use) then you may need to contact IBM or jClarity or provide a patch yourself.

**Comment by JScheiterbauer** [comment]
@karianna 
ok, 8u222 will be soon enough, thank you :)

**Comment by jt214** [comment]
I created another PR #1113 for 64-but builds.
